tools/serverpod_cli/lib/src/generator/dart/library_generators/model_library_generator.dart

```diff
@@ -432,13 +433,12 @@ class SerializableModelLibraryGenerator {
           noneMutableTypeNames.contains(field.type.className))) {
         assignment = refer('this').property(field.name);
       } else if (clonableTypeNames.contains(field.type.className)) {
-        assignment = _buildMaybeNullMethodCall(field, 'clone');
+        assignment = _buildClonableCopyWithAssignment(field);
```

Comment:
Suggestion: Branch based on what should be built here instead of inside of `_buildClonableCopyWithAssignment`.

By making this code a bit more shallow it becomes easier to follow.

Reply:
We could call `_buildDeepCloneTree` on this abstraction level for the related classes.

Reply:
Happy to make the change but I personally it's not obvious to me that it would make it more readable. It would look like this:
```dart
if ((field.type.isEnumType ||
      noneMutableTypeNames.contains(field.type.className))) {
    assignment = refer('this').property(field.name);
} else if (clonableTypeNames.contains(field.type.className)) {
  var shouldDeepClone = field.type.isListType || field.type.isMapType;
  assignment = switch (shouldDeepClone) {
    true => _buildDeepCloneTree(
        field.type,
        field.name,
        root: true,
      ),
    false => refer(Keyword.strictShallowCloneFunctionName,
            serverpodUrl(serverCode))
        .call([refer(field.name)]),
  };
} else {
  assignment = _buildMaybeNullMethodCall(field, 'copyWith');
}
```
- In general I believe nested code is harder to follow and especially if you want to quickly skim the code for the relevant parts. When skimming I want to quickly find the relevant branch (i.e. enum, clonable type or else. Having too much noise makes this search harder.
- I also believe it helps with readability when different parts of a function body has roughly the same branch depth/complexity. When one part sticks out like this I think it can be a hint that it should be broken out.

I do agree that _too_ many layers of abstractions makes code hard to read as well, especially in class hierarchies. But for class methods which are easy to jump around between I generally tolerate a bit more.

Again, happy to make the change but would be curious to hear your thoughts on my points above, both in general and in this specific case!

Reply:
I think it would make more sense to keep the if statement shallow. That is, add the check for isListType and isMapType to the outer if statement.

Reply:
Apologies, I missed that it was possible to simply flatten it out to the top level if/else. Will fix, thanks! 👍